ngraph/test/type_prop/interpolate.cpp

```diff
@@ -0,0 +1,130 @@
+    attrs.cube_coeff = -0.75;
+    auto interp = std::make_shared<op::v4::Interpolate>(image, target_shape, scales, axes, attrs);
+
+    EXPECT_EQ(interp->get_element_type(), element::f32);
+    EXPECT_EQ(interp->get_shape(), (Shape{2, 2, 15, 30}));
+}
+
+TEST(type_prop, interpolate_v4_partial)
+{
+    auto partial_shape = PartialShape{2, 2, Dimension::dynamic(), Dimension::dynamic()};
+
+    auto image = std::make_shared<op::Parameter>(element::f32, partial_shape);
+    auto target_shape = std::make_shared<op::Parameter>(element::f32, Shape{2, 2, 15, 30});
+    auto scales = op::Constant::create<float>(element::f32, Shape{2}, {0.5f, 0.5f});
+    auto axes = op::Constant::create<int64_t>(element::i64, Shape{2}, {2, 3});
+
+    InterpolateAttrs attrs;
+    attrs.mode = InterpolateMode::nearest;
+    attrs.shape_calculation_mode = ShapeCalcMode::scales;
+    attrs.coordinate_transformation_mode = CoordinateTransformMode::half_pixel;
+    attrs.nearest_mode = Nearest_mode::round_prefer_floor;
+    attrs.antialias = false;
+    attrs.pads_begin = {0, 0, 0, 0};
+    attrs.pads_end = {0, 0, 0, 0};
+    attrs.cube_coeff = -0.75;
+    auto interp = std::make_shared<op::v4::Interpolate>(image, target_shape, scales, axes, attrs);
+
+    EXPECT_EQ(interp->get_element_type(), element::f32);
+    ASSERT_TRUE(interp->get_output_partial_shape(0).same_scheme(partial_shape));
+
+    // rank unknown
+    auto partial_param = std::make_shared<op::Parameter>(element::f32, PartialShape::dynamic());
+    auto interp_part =
+        std::make_shared<op::v4::Interpolate>(partial_param, target_shape, scales, axes, attrs);
+    ASSERT_TRUE(interp_part->get_output_partial_shape(0).same_scheme(PartialShape::dynamic()));
+}
+
+TEST(type_prop, interpolate_v4_partial_static_rank)
+{
+    auto partial_shape = PartialShape{2, 2, Dimension::dynamic(), Dimension::dynamic()};
+
+    auto image = std::make_shared<op::Parameter>(element::f32, partial_shape);
+    auto target_shape = std::make_shared<op::Parameter>(element::f32, Shape{2, 2, 15, 30});
+    auto scales = op::Constant::create<float>(element::f32, Shape{2}, {0.5f, 0.5f});
+    auto axes = op::Constant::create<int64_t>(element::i64, Shape{2}, {2, 3});
+
+    InterpolateAttrs attrs;
+    attrs.mode = InterpolateMode::nearest;
+    attrs.shape_calculation_mode = ShapeCalcMode::scales;
+    attrs.coordinate_transformation_mode = CoordinateTransformMode::half_pixel;
+    attrs.nearest_mode = Nearest_mode::round_prefer_floor;
+    attrs.antialias = false;
+    attrs.pads_begin = {0, 0, 0, 0};
+    attrs.pads_end = {0, 0, 0, 0};
+    attrs.cube_coeff = -0.75;
+    auto interp = std::make_shared<op::v4::Interpolate>(image, target_shape, scales, axes, attrs);
+
+    EXPECT_EQ(interp->get_element_type(), element::f32);
+    ASSERT_TRUE(interp->get_output_partial_shape(0).same_scheme(partial_shape));
+    ASSERT_TRUE(interp->get_output_partial_shape(0).rank().is_static());
+}
+
+TEST(type_prop, interpolate_v4_partial_static_rank2)
```

Comment:
Need to add a test to cover case described in ticket 35166. For example, scale = [1/3], input shape = [3]. The output shape should be [1] (not [0]!).

Reply:
Done.